packages/story-editor/src/app/media/utils/useProcessMedia.js

```diff
@@ -478,11 +482,86 @@ function useProcessMedia({
     ]
   );
 
+  /**
+   * Crop hidden video using FFmpeg.
+   *
+   * @param {import('@googleforcreators/media').Resource} resource Resource object.
+   */
+  const cropHiddenVideo = useCallback(
+    ({ id: elementId, resource: oldResource }, cropParams) => {
+      const { id: resourceId, src: url, mimeType } = oldResource;
+      const { newWidth, newHeight, cropElement } = cropParams;
+
+      const onUploadSuccess = ({ id, resource }) => {
+        copyResourceData({ oldResource, resource });
+        updateOldTranscodedObject(oldResource.id, resource.id, 'source-image');
+
+        updateElementById({
+          elementId,
+          properties: {
+            x: cropElement.x < 0 ? 0 : cropElement.x,
+            y:
+              cropElement.y < DANGER_ZONE_HEIGHT
+                ? -DANGER_ZONE_HEIGHT
+                : cropElement.y,
+            width: newWidth,
+            height: newHeight,
+          },
+        });
+
+        // onUploadSuccess is also called with previousResourceId,
+        // for which we don't need to run this.
+        if (id === resource.id) {
+          postProcessingResource(resource);
+        }
+      };
+
+      const onUploadProgress = ({ resource }) => {
+        const oldResourceWithId = { ...resource, id: oldResource.id };
+        updateExistingElementsByResourceId(resourceId, {
+          ...oldResourceWithId,
+        });
```

Comment:
This might be better done in the `onUploadSuccess` callback.

Otherwise what can be observed is that updating the resource & updating the element's dimensions happens in two steps, which is odd.

Of course we can also update the element's dimensions in the `onUploadProgress` callback for faster feedback loop.

But then we also need to restore the original resource & dimensions in `onUploadError`. Otherwise when there's an error with video cropping/uploading, it will lead to a broken appearance on canvas:

![Screenshot 2022-08-31 at 12 34 53](https://user-images.githubusercontent.com/841956/187660114-f1c1eaad-d744-495c-b597-c22727604427.png)

-----

Aside: we only want to update this specific element, not all elements with this resource. So we shouldn't use `updateExistingElementsByResourceId`.

Reply:
Okay that makes sense --- will update using `onUploadSuccess`